### PR TITLE
docs: move to table of features for `leptos`

### PR DESCRIPTION
**Pull request by chrisp60**
This is a cosmetic change and may be opinionated. I find tables like this to be more readable.

For some it can be hard to parse without visual organization. No issue at all if it does not fall into line with your view for the docs.

I would not mind unifying the other crates if merged

here it is rendered on my local machine without any styles that docs.rs might add.
image

### CODE REVIEW

File: leptos/src/lib.rs [hunk 1]
```diff
@@ -140,6 +118,37 @@
...
+//! | serde                     | **(De)Serialization**: use [serde] for [Resource]s.           | Yes     |
+//! | serde-lite                | **(De)Serialization**: use [serde-lite] for [Resource]s.      | No      |
+//! | rkyv                      | **(De)Serialization**: use [rkyv] for [Resource]s.            | No      |
+//! | miniserde                 | **(De)Serialization**: use [miniserde] for [Resource]s.       | No      |
+//! | tracing                   | **Logs & Telemetry**, [tracing] support for [component]s.     | No      |
+//! | default-tls[^2]           | **Networking**: Use default native TLS support.               | No      |
+//! | rustls[^2]                | **Networking**: Use [rustls].                                 | No      |
+//! | template_macro            | **Templating**: Enables the legacy [template] macro.          | No      |
+//!
+//! [^1]: At least one **rendering** feature must be enabled. An Isomorphic crate may have
```
Comment (gbj) [inline leptos/src/lib.rs:140]: This rewritten version sounds misleading to me. Exactly one should be enabled, not more than one.

**Comment by gbj** [comment]
I like this overall, and thank you for it! I would prefer to keep the original wording about using **one** of `csr`, `ssr`, or `hydrate`.

**Comment by chrisp60** [comment]
Hey! I see your recommendations and you are correct. I'm a little overwhelmed with work at the moment but plan to revisit and correct this when I have the time. Thanks for the feedback. 

I am going to move this to a draft for now.
